bot/utils/helpers.py

```diff
@@ -0,0 +1,71 @@
+import asyncio
+from typing import Any, Callable, Iterable, Optional, TypeVar
+
+import discord
+import more_itertools
+
+from bot import logger
+
+T = TypeVar('T')
+
+
+async def chunked_find(
+    predicate: Callable[[T], Any],
+    seq: Iterable[T],
+    *,
+    chunk_size: Optional[int] = None
+) -> Optional[T]:
+    """
+    A helper to return the first element found in the sequence that meets the predicate.
+
+    If chunk_size is specified, chunked_find() will yield control to the event loop between each chunk.
+    """
+    for chunk in more_itertools.chunked(seq, chunk_size):
+        for element in chunk:
+            if predicate(element):
+                return element
+
+        if chunk_size:
+            await asyncio.sleep(0)  # Yield to the event loop
+    return None
+
+
+async def _check_first_message_referencing(thread: discord.Thread, message_id: int) -> Optional[discord.Thread]:
+    """Returns a thread if the thread_starter_message in `thread` references the message_id."""
```

Comment:
I realize that if we're here, we can simply return a bool.
```suggestion
    """Return a bool indicating if the thread starter message references the given ``message_id`."""
```